contrib/pipelines/core/store_s3.go

```diff
@@ -124,14 +110,14 @@ func (s *Storage) StoreFlows(flows []*flow.Flow) error {
 	// transform flows and save to in-memory arrays, based on tag
 	if flows != nil {
 		for _, fl := range flows {
-			flowTag := s.flowClassifier.GetFlowTag(fl)
-			if _, ok := s.excludedTags[flowTag]; ok {
+			flowTag := s.pipeline.Classifier.GetFlowTag(fl)
+			if s.pipeline.Filterer.IsExcluded(flowTag) {
 				continue
 			}
 
 			var transformedFlow interface{}
-			if s.flowTransformer != nil {
-				transformedFlow = s.flowTransformer.Transform(fl)
+			if s.pipeline.Transformer != nil && !reflect.ValueOf(s.pipeline.Transformer).IsNil() {
```

Comment:
reflection is usually not good for performances, I would get rid of this or at least could be moved outside of the loop

Reply:
removed the entire if ... we currently can assume we have a transformer ... in future I can just implement a null transformator which does not do any transformation and that can be used in-place of `nil`